tests/span_ext_tests.cpp

```diff
@@ -1,360 +1,348 @@
-         EXPECT_TRUE(s.data() == std::addressof(arr2d[0]));
-     }
-
-     {
-         const auto s = make_span(std::addressof(arr3d[0]), 1);
-         EXPECT_TRUE(s.size() == 1);
-         EXPECT_TRUE(s.data() == std::addressof(arr3d[0]));
-     }
- }
-
- TEST(span_ext_test, make_span_from_dynamic_array_constructor)
- {
-     double(*arr)[3][4] = new double[100][3][4];
-
-     {
-         auto s = make_span(&arr[0][0][0], 10);
-         EXPECT_TRUE(s.size() == 10);
-         EXPECT_TRUE(s.data() == &arr[0][0][0]);
-     }
-
-     delete[] arr;
- }
-
- TEST(span_ext_test, make_span_from_std_array_constructor)
- {
-     std::array<int, 4> arr = {1, 2, 3, 4};
-
-     {
-         auto s = make_span(arr);
-         EXPECT_TRUE(s.size() == arr.size());
-         EXPECT_TRUE(s.data() == arr.data());
-     }
-
-     // This test checks for the bug found in gcc 6.1, 6.2, 6.3, 6.4, 6.5 7.1, 7.2, 7.3 - issue #590
-     {
-         span<int> s1 = make_span(arr);
-
-         static span<int> s2;
-         s2 = s1;
-
- #if defined(__GNUC__) && __GNUC__ == 6 && (__GNUC_MINOR__ == 4 || __GNUC_MINOR__ == 5) &&          \
-     __GNUC_PATCHLEVEL__ == 0 && defined(__OPTIMIZE__)
-         // Known to be broken in gcc 6.4 and 6.5 with optimizations
-         // Issue in gcc: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=83116
-         EXPECT_TRUE(s1.size() == 4);
-         EXPECT_TRUE(s2.size() == 0);
- #else
-         EXPECT_TRUE(s1.size() == s2.size());
- #endif
-     }
- }
-
- TEST(span_ext_test, make_span_from_const_std_array_constructor)
- {
-     const std::array<int, 4> arr = {1, 2, 3, 4};
-
-     {
-         auto s = make_span(arr);
-         EXPECT_TRUE(s.size() == arr.size());
-         EXPECT_TRUE(s.data() == arr.data());
-     }
- }
-
- TEST(span_ext_test, make_span_from_std_array_const_constructor)
- {
-     std::array<const int, 4> arr = {1, 2, 3, 4};
-
-     {
-         auto s = make_span(arr);
-         EXPECT_TRUE(s.size() == arr.size());
-         EXPECT_TRUE(s.data() == arr.data());
-     }
- }
-
- TEST(span_ext_test, make_span_from_container_constructor)
- {
-     std::vector<int> v = {1, 2, 3};
-     const std::vector<int> cv = v;
-
-     {
-         auto s = make_span(v);
-         EXPECT_TRUE(s.size() == v.size());
-         EXPECT_TRUE(s.data() == v.data());
-
-         auto cs = make_span(cv);
-         EXPECT_TRUE(cs.size() == cv.size());
-         EXPECT_TRUE(cs.data() == cv.data());
-     }
- }
-
- TEST(span_test, interop_with_gsl_at)
- {
-     int arr[5] = {1, 2, 3, 4, 5};
-     span<int> s{arr};
-     EXPECT_TRUE(at(s, 0) ==  1);
-     EXPECT_TRUE(at(s, 1) == 2);
- }
-
- TEST(span_ext_test, iterator_free_functions)
- {
-     int a[] = {1, 2, 3, 4};
-     span<int> s{a};
-
-     EXPECT_TRUE((std::is_same<decltype(s.begin()), decltype(begin(s))>::value));
-     EXPECT_TRUE((std::is_same<decltype(s.end()), decltype(end(s))>::value));
-
-     EXPECT_TRUE((std::is_same<decltype(s.cbegin()), decltype(cbegin(s))>::value));
```

Comment:
If you do bring back the `span_ext` functions as I suggest, these could return in the form `EXPECT_TRUE((std::is_same<decltype(cbegin(s)), decltype(std::cbegin(s))>::value));` etc.